pkg/ipam/core/node.go

```diff
@@ -221,25 +237,30 @@ func (n *node) allocateIPWithAddr(addr netip.Addr) *netip.Addr {
 	}
 
 	for i := range n.ips {
-		if n.ips[i].Compare(addr) == 0 {
+		if n.ips[i].addr.Compare(addr) == 0 {
 			return nil
 		}
 	}
 
-	n.ips = append(n.ips, addr)
+	n.ips = append(n.ips, nodeIP{addr: addr, creationTimestamp: time.Now()})
+	n.lastUpdateTimestamp = time.Now()
 
-	return &n.ips[len(n.ips)-1]
+	return &n.ips[len(n.ips)-1].addr
```

Comment:
Not related to this PR, but should't we set this as lastIPs? Or we don't want to advance to prevent eventual holes?

Reply:
Simply the 2 allocation types are not related